contracts/AbstractWasabiPoolV2.sol

```diff
@@ -0,0 +1,632 @@
+                }
+                optionIdsForToken.add(optionId);
+            }
+        }
+        optionIds.add(optionId);
+        idToFilledOrCancelled[_request.id] = true;
+
+        emit OptionIssued(optionId, _request.premium, _request.id);
+        return optionId;
+    }
+
+    /// @inheritdoc IWasabiPoolV2
+    function writeOption(
+        WasabiStructsV2.PoolAsk calldata _request, bytes calldata _signature
+    ) external payable returns (uint256) {
+        return writeOptionTo(_request, _signature, _msgSender());
+    }
+
+    /**
+     * @dev Validates the given PoolAsk in order to issue an option
+     */
+    function validate(WasabiStructsV2.PoolAsk calldata _request, bytes calldata _signature) internal {
+        // 1. Validate Signature
+        address signer = PoolAskVerifierV2.getSignerForPoolAsk(_request, _signature);
+        if (signer == address(0) || (signer != admin && signer != owner())) {
+            revert IWasabiErrors.InvalidSignature();
+        }
+
+        // 2. Validate Meta
+        if (_request.orderExpiry < block.timestamp) {
+            revert IWasabiErrors.HasExpired();
+        }
+
+        require(_request.poolAddress == address(this), "WasabiPool: Signature doesn't belong to this pool");
+        validateAndWithdrawPayment(_request.premium, "WasabiPool: Not enough premium is supplied");
+
+        // 3. Request Validation
+        if (_request.strikePrice == 0) {
+            revert IWasabiErrors.InvalidStrike();
+        }
+        if (_request.expiry == 0) {
+            revert IWasabiErrors.InvalidExpiry();
+        }
+
+        // 4. Type specific validation
+        if (_request.optionType == WasabiStructsV2.OptionType.CALL) {
+
+            if (!isAvailableTokenId(_request.collection, _request.tokenId)) {
+                revert IWasabiErrors.NftIsInvalid();
+            }
+
+            if (isERC721(_request.collection)) {
+                uint256 optionId = erc721TokenToOptions[abi.encodePacked(_request.collection, _request.tokenId)];
+                if (isValid(optionId) && IERC721(_request.collection).ownerOf(_request.tokenId) != owner()) {
+                    revert IWasabiErrors.NftIsInvalid();         
+                }
+            } else if (isERC1155(_request.collection)) {
+                EnumerableSet.UintSet storage optionIdsForToken = erc1155TokenToOptions[abi.encodePacked(_request.collection, _request.tokenId)];
+                
+                bool isUnlocked;
+                for (uint256 i = 0; i < optionIdsForToken.length(); i++) {
+                    if (!isValid(optionIdsForToken.at(i))) {
+                        isUnlocked = true;
+                        break;
+                    }
+                }
+                if (IERC1155(_request.collection).balanceOf(owner(), _request.tokenId) == 0 && !isUnlocked) {
+                    revert IWasabiErrors.NftIsInvalid();
+                }
+            }
```

Comment:
The reason I added above code is:
isAvailableTokenId function is checks only if pool or owner has certain tokenId.
But even pool has tokenId, if it has valid option, validation should be fail.
That's why I added the code.
Of course, tx will be failed after validation function if pool has valid option because we are checking again.